src/internal/EnhancedSwitch.js

```diff
@@ -13,15 +13,15 @@ function getStyles(props, context) {
   return {
     root: {
       position: 'relative',
-      cursor: props.disabled ? 'default' : 'pointer',
+      cursor: props.disabled ? 'not-allowed' : 'pointer',
       overflow: 'visible',
       display: 'table',
       height: 'auto',
       width: '100%',
     },
     input: {
       position: 'absolute',
-      cursor: props.disabled ? 'default' : 'pointer',
+      cursor: props.disabled ? 'not-allowed' : 'pointer',
```

Comment:
@nathanmarks What do you mean? I think that's pretty much the same conversation as for the `c/internal/EnhancedButton.js`.
@nktssh Otherwise, what about spliting this PR into two chunks, so we can merge some of those changes?